backend/app/db/crud/litigation.py

```diff
@@ -0,0 +1,231 @@
+import logging
+from typing import Any, Mapping, Set
+from app.db.models.litiguation import LitParty, LitPartyType, LitSideType
+from app.db.models.case import (
+    Case,
+    CaseStatus,
+    ClimateAlignmentClass,
+    StrategicAlignmentClass,
+    UNFCCCPillars,
+)
+from app.db.models.geography import Geography
+from app.db.models.document import Sector
+
+from sqlalchemy.orm import Session
+
+
+def _to_bool(value: str) -> bool:
+    return value.upper() == "YES" or value.upper() == "TRUE"
+
+
+def _validate_refs(
+    log: logging.Logger,
+    entities: Mapping[str, Any],
+    entity_name: str,
+    entity_key: str,
+    primary_set: Set[str],
+):
+    log.info(f"Checking {entity_name}:")
+    referenced_cases = set([e[entity_key] for e in entities.values()])
+    unknown_cases = referenced_cases - primary_set
+    orphaned_cases = primary_set - referenced_cases
+
+    log.info(f"  Found {len(referenced_cases)} referenced {entity_key}s.")
+
+    log.info(f"  Got {len(unknown_cases)} unknown referenced {entity_key}s.")
+    if len(unknown_cases) > 0:
+        log.info(unknown_cases)
+
+    log.info(
+        f"  Leaving {len(orphaned_cases)} {entity_key}(s) orphaned of {entity_name}."
+    )
+    if len(orphaned_cases) > 0:
+        log.info(orphaned_cases)
+
+
+def verify(
+    log: logging.Logger,
+    cases: Mapping[str, Mapping],
+    events: Mapping[str, Mapping],
+    documents: Mapping[str, Mapping],
+    parties: Mapping[str, Mapping],
+):
+    """Checks the relationships
+
+    Event:      Case ID
+    Parties     Case ID
+    Document:   Case ID     Event ID
+    """
+
+    all_cases = set(cases.keys())
+    log.info(f"Found {len(cases)} Cases, {len(all_cases)} unique.")
+
+    _validate_refs(log, events, "Events", "Case ID", all_cases)
+    _validate_refs(log, parties, "Parties", "Case ID", all_cases)
+    _validate_refs(log, documents, "Documents", "Case ID", all_cases)
+    _validate_refs(log, documents, "Documents", "Event ID", set(events))
+
+
+def transform_to_json(
+    log: logging.Logger,
+    cases: Mapping[str, Mapping],
+    events: Mapping[str, Mapping],
+    documents: Mapping[str, Mapping],
+    parties: Mapping[str, Mapping],
+):
+    json_cases = {}
+
+    # Initalise attributes
+    for idx, cid in enumerate(cases):
+        json_cases[cid] = dict(cases[cid])
+        json_cases[cid]["events"] = []
+        json_cases[cid]["documents"] = []
+        json_cases[cid][LitSideType.FILING_PARTY.value] = []
+        json_cases[cid][LitSideType.RESPONDING_PARTY.value] = []
+        json_cases[cid][LitSideType.INTERVENOR.value] = []
+
+    # Add events
+    for _, e in events.items():
+        cid = e["Case ID"]
+        if cid not in json_cases.keys():
+            msg = f"Event {e} references a non-existent case {cid}"
+            log.error(msg)
+            # raise ValueError(f"Event {e} references a non-existent case {cid}")  - uncomment to abort instead
```

Comment:
Moving fast at the minute so raising exceptions should be put back before release